packages/snaps-execution-environments/src/webview/ProxyMessageStream.test.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * @jest-environment jsdom
+ */
+/* eslint-disable @typescript-eslint/consistent-type-definitions */
+import { ProxyMessageStream } from './ProxyMessageStream';
+
+describe('ProxyMessageStream', () => {
+  let originalPostMessage: any;
+
+  beforeEach(() => {
+    originalPostMessage = window.postMessage;
+
+    Object.assign(global, {
```

Comment:
I don't think we need to pollute the global scope for these tests? We can just pass in a mocked window?